typeobjectpattern/src/main/java/com/iluwatar/typeobject/CandyGame.java

```diff
@@ -0,0 +1,171 @@
+  
+  CandyGame(int num, CellPool pool) {
+    this.cells = new Cell[num][num];
+    this.pool = pool;
+    this.totalPoints = 0;
+    for (int i = 0; i < num; i++) {
+      for (int j = 0; j < num; j++) {
+        this.cells[i][j] = this.pool.getNewCell();
+        this.cells[i][j].xIndex = j;
+        this.cells[i][j].yIndex = i;
+      }
+    }
+  }
+  
+  static String numOfSpaces(int num) {
+    String result = "";
+    for (int i = 0; i < num; i++) {
+      result += " ";
+    }
+    return result;
+  }
+  
+  void printGameStatus() {
+    System.out.println("");
+    for (int i = 0; i < cells.length; i++) {
+      for (int j = 0; j < cells.length; j++) {
+        String candyName = cells[i][j].candy.name;
+        if (candyName.length() < 20) {
+          int totalSpaces = 20 - candyName.length();
+          System.out.print(numOfSpaces(totalSpaces / 2) + cells[i][j].candy.name
+              + numOfSpaces(totalSpaces - totalSpaces / 2) + "|");
+        } else {
+          System.out.print(candyName + "|");
+        }
+      }
+      System.out.println("");
+    }
+    System.out.println("");
+  }
+  
+  ArrayList<Cell> adjacentCells(int yIndex, int xIndex) {
+    ArrayList<Cell> adjacent = new ArrayList<Cell>();
+    if (yIndex == 0) {
+      adjacent.add(this.cells[1][xIndex]);
+    }
+    if (xIndex == 0) {
+      adjacent.add(this.cells[yIndex][1]);
+    }
+    if (yIndex == cells.length - 1) {
+      adjacent.add(this.cells[cells.length - 2][xIndex]);
+    }
+    if (xIndex == cells.length - 1) {
+      adjacent.add(this.cells[yIndex][cells.length - 2]);
+    }
+    if (yIndex > 0 && yIndex < cells.length - 1) {
+      adjacent.add(this.cells[yIndex - 1][xIndex]);
+      adjacent.add(this.cells[yIndex + 1][xIndex]);
+    }
+    if (xIndex > 0 && xIndex < cells.length - 1) {
+      adjacent.add(this.cells[yIndex][xIndex - 1]);
+      adjacent.add(this.cells[yIndex][xIndex + 1]);
+    }
+    return adjacent;
+  }
+   
+  boolean continueRound() {
+    for (int i = 0; i < this.cells.length; i++) {
+      if (this.cells[cells.length - 1][i].candy.getType().equals(Type.rewardFruit)) {
+        return true;
+      }
+    }
+    for (int i = 0; i < this.cells.length; i++) {
+      for (int j = 0; j < this.cells.length; j++) {
+        if (!this.cells[i][j].candy.getType().equals(Type.rewardFruit)) {
+          ArrayList<Cell> adj = adjacentCells(i,j);
+          for (int a = 0; a < adj.size(); a++) {
+            if (this.cells[i][j].candy.name.equals(adj.get(a).candy.name)) {
+              return true;
+            }
+          }
+        }
+      }
+    }
+    return false;
+  }
+  
+  void handleChange(int points) {
+    System.out.println("+" + points + " points!");
+    this.totalPoints += points; 
+    printGameStatus();
+  }
+  
+  void round(int timeSoFar, int totalTime) {
+    long start = System.currentTimeMillis();
+    long end = System.currentTimeMillis();
+    while (end - start + timeSoFar < totalTime && continueRound()) {
```

Comment:
same here. `start` and `end` are essentially the same.

Reply:
It's needed so that we can enter into while loop